gcd/src/main/java/org/spine3/server/storage/datastore/IdTransformer.java

```diff
@@ -0,0 +1,197 @@
+        if (parametrizedClass == null) {
+            return tryAllSupportedClasses(stringId);
+        }
+
+        try {
+            @SuppressWarnings("unchecked")  // cast should be safe, since the convention.
+            final Class<I> result = Classes.getGenericParameterType(parametrizedClass, 0);
+            return result;
+        } catch (ClassCastException e) {
+            LOG.warning(UNABLE_TO_DETECT_GENERIC_TYPE + e.toString());
+            return tryAllSupportedClasses(stringId);
+        }
+    }
+
+    @SuppressWarnings({"unchecked", "ResultOfMethodCallIgnored"})
+    private static <I> Class<I> tryAllSupportedClasses(String stringId) {
+        if (stringId.startsWith(TYPE_PREFIX) && stringId.endsWith(SERIALIZED_MESSAGE_BYTES_POSTFIX)) {
+            return (Class<I>) Message.class;
+        } else {
+            try {
+                Long.parseLong(stringId);
+                return (Class<I>) Long.class;
+            } catch (NumberFormatException ignored) {
+                return (Class<I>) String.class;
+            }
+        }
+    }
+
+    private static Message protoIdFromString(String stringId) {
+        checkArgument(stringId.startsWith(TYPE_PREFIX), String.format(WRONG_OR_BROKEN_MESSAGE_ID, stringId));
+        checkArgument(stringId.endsWith(SERIALIZED_MESSAGE_BYTES_POSTFIX), String.format(WRONG_OR_BROKEN_MESSAGE_ID, stringId));
+
+        final int typeStartIndex = stringId.indexOf(TYPE_PREFIX) + TYPE_PREFIX.length();
+        final int typeEndIndex = stringId.indexOf(SERIALIZED_MESSAGE_DIVIDER);
+        final String typeName = stringId.substring(typeStartIndex, typeEndIndex);
+
+        final int dataStartIndex = stringId.indexOf(SERIALIZED_MESSAGE_DIVIDER) + SERIALIZED_MESSAGE_DIVIDER.length();
+        final int dataEndIndex = stringId.lastIndexOf(SERIALIZED_MESSAGE_BYTES_POSTFIX);
+        final String bytesString = stringId.substring(dataStartIndex, dataEndIndex);
+
+        if (bytesString.equals(SERIALIZED_DEFAULT_MESSAGE)) {
+            final TypeUrl typeUrl = TypeUrl.of(typeName);
+            final Message id = Entities.defaultMessage(typeUrl);
+            return id;
+        }
+
+        final String[] separateStringBytes = bytesString.split(SERIALIZED_MESSAGE_BYTES_DIVIDER);
+        final byte[] messageBytes = new byte[separateStringBytes.length];
+        for (int i = 0; i < messageBytes.length; i++) {
+            final byte oneByte = Byte.parseByte(separateStringBytes[i], SERIALIZED_BYTES_RADIX);
+            messageBytes[i] = oneByte;
+        }
+
+        final ByteString byteString = ByteString.copyFrom(messageBytes);
+        final Any wrappedId = Any.newBuilder()
+                .setValue(byteString)
+                .setTypeUrl(typeName)
+                .build();
+        final Message id = AnyPacker.unpack(wrappedId);
+        return id;
+    }
+
+
+    private static boolean isNumber(Class<?> clss) {
```

Comment:
The name is wrong. `isNumber` means `capture<? extends Number>`. 
I'd suggest `isOfSupportedNumberType`.

Instead of `clss` please use `clazz`, which is a common replacement for a keyword.

Also the implementation is wrong. `double` is a primitive. But I don't think that's what you want to return `true`.